Add CMakeLists file for CMake build

JaLooNz opened the pull request:
- Add CMakeLists file for CMake build for libraries and applications
- Various options are disabled instead of enable
- ECDH functionality will break self-test application

JaLooNz committed:
Add CMakeLists file for CMake build
-Add CMakeLists file for CMake build for libraries and applications
-Various options are disabled instead of enable
-ECDH functionality will break self-test application

rben-dev commented:
Hi @JaLooNz 

Thanks for your interest in libecc and your PR!

I have small remarks before merging this:
- First, I cannot see the self tests binaries produced after `cmake` and `make` are called. Only the static libraries are built. I might have missed something?
- Also, as I can see the `PLATFORM_WORD_SIZE` is always set to 32, but the best option would be if it is not defined to let the library decide itself (as in the header a self detection is made to fit the compilation platform with the best size).

Thanks in advance,

JaLooNz committed:
-Default enable all algorithms (except some GOST ciphers)
-DEfault build ec-self-tests and ec-utils

JaLooNz committed:
-Remove PLATFORM_WORD_SIZE configuration
-Add detection for CMAKE_GENERATOR and CMAKE_CL_64 (working for x86 and x86_64 windows builds using Visual Studio)

JaLooNz commented:
Hi,
CMakeLists updated to address the above issues.
Additionally, also fixed ECDH options due to missing source for executable.
